src/libAtomVM/opcodesswitch.h

```diff
@@ -108,14 +108,17 @@ typedef struct
 
 typedef dreg_t dreg_gc_safe_t;
 
+#define DEST_REGISTER(reg) dreg_t reg
+#define GC_SAFE_DEST_REGISTER(reg) dreg_gc_safe_t reg
+
 #define T_DEST_REG(dreg) \
     reg_type_c((dreg).reg_type), (int) ((dreg).index)
 
 #define T_DEST_REG_GC_SAFE(dreg) T_DEST_REG(dreg)
 #else
 
```

Comment:
Not needed for this PR, but It would be good to get in the habit of commenting which `#if` the `#else` and `#endif` lines refer to, especially with nested `#if`’s and the fact that we are not using indentation for these macros.

Reply:
Agreed.